spec/index.md

```diff
@@ -1615,6 +1617,20 @@ Example: To accept all cycles provided in a call, invoke `ic0.msg_cycles_accept(
 
     This does not trap.
 
+-   `ic0.burn_cycles : (amount_to_burn : i64) → (amount : i64)`
+
+    This burns cycles from the canister. It burns as many cycles as possible, up to these constraints:
+
+    It burns no more cycles than `amount_to_burn`.
+
+    It moves no more cycles than available according to `balance` - `freezing_threshold`, and
```

Comment:
`freezing_threshold` is a value in time units how long the canister should remain installed: we can't deduct time duration from cycles; instead, we need to compute the so-called freezing limit in cycles. I'd reformulate the condition to saying that the canister doesn't become frozen after this system API call.

Reply:
I am not sure that formulation is correct, as far as I understand canister may become frozen if it has `freezing_limith` cycles left. But it will not be deleted, since it will use those resources to pay for memory and other fixed costs.

Reply:
Made a concrete edit suggestions => closing this thread.